var/spack/repos/builtin/packages/libint/package.py

```diff
@@ -244,3 +260,73 @@ def patch(self):
             filter_file(
                 "$(CXX) $(CXXFLAGS)", "$(FC) $(FCFLAGS)", "export/fortran/Makefile", string=True
             )
+
+
+class CMakeBuilder(
+    # Declare autotools first, so that building the libint compiler is very
+    # to the old process
+    spack.build_systems.autotools.AutotoolsBuilder,
+    spack.build_systems.cmake.CMakeBuilder,
+):
+    phases = ("autoreconf", "configure", "build", "cmake", "cmake_build", "install")
+
+    force_autoreconf = True
+
+    root_cmakelists_dir = "generated"
+
+    def autoreconf(self, pkg, spec, prefix):
+        which("bash")("autogen.sh")
+        # skip tarball creation and removal of dir with generated code
+        # The regexp is different from @2.6.0 since export/Makefile changed
+        # TODO:: More robust solution
+        filter_file(r"^(export::.*)\s+tgz( .*)$", r"\1\2", "export/Makefile")
+
+    @property
+    def optflags(self):
+        flags = "-O2"
+
+        # microarchitecture-specific optimization flags should be controlled
+        # by Spack, otherwise we may end up with contradictory or invalid flags
+        # see https://github.com/spack/spack/issues/17794
+
+        return flags
+
+    def setup_build_environment(self, env):
+        # Set optimization flags
+        env.set("CFLAGS", self.optflags)
+        env.set("CXXFLAGS", self.optflags)
+
+        # Change AR to xiar if we compile with Intel and we
+        # find the executable
+        if "%intel" in self.spec and which("xiar"):
+            env.set("AR", "xiar")
+
+    def configure_args(self):
+        return define_configure_args(self.pkg, self.spec, self.prefix, self.optflags)
```

Comment:
are we sure that these make it into the cmake part of the build?

(just a question from someone who does not understand what exactly happens here, maybe it's all fine)

Reply:
Not really. I'm also a bit unsure what exactly happens with the double inheritance. That's where I wanted to look next (but haven't got much time yet).